library/std/src/primitive_docs.rs

```diff
@@ -804,10 +804,12 @@ mod prim_tuple {}
 /// often discard insignificant digits: `println!("{}", 1.0f32 / 5.0f32)` will
 /// print `0.2`.
 ///
-/// Additionally, `f32` can represent a couple of special values:
+/// Additionally, `f32` can represent some special values:
 ///
-/// - `-0`: this is just due to how floats are encoded. It is semantically
-///   equivalent to `0` and `-0.0 == 0.0` results in `true`.
+/// - `-0`: this value exists due to how floats are encoded. -0 == 0 is true, but for other
```

Comment:
how about: `-0 == 0 is true, but for some other`

Reply:
Also, the . is important I think to make sure it's understood that one is talking about floats.

Reply:
It seemed odd to have it be 0 in one spot and 0.0 in another spot, and currently we do print 0 or -0 here, and it has context.

Reply:
The way I'd explain it, `-0.0 == 0.0` refers to Rust code. `-0 == 0` would mean integer comparison. I pasted it into the playground when I rad that comment to test it out and got an error because I hadn't added the dot.

Reply:
That's fair, let's make it consistently 0.0 then.